src/rgw/rgw_http_client.cc

```diff
@@ -832,16 +836,18 @@ int RGWHTTPManager::process_requests(bool wait_for_data, bool *done)
     CURLMsg *msg;
     while ((msg = curl_multi_info_read((CURLM *)multi_handle, &msgs_left))) {
       if (msg->msg == CURLMSG_DONE) {
-	CURL *e = msg->easy_handle;
-	rgw_http_req_data *req_data;
-	curl_easy_getinfo(e, CURLINFO_PRIVATE, (void **)&req_data);
-
-	long http_status;
-	curl_easy_getinfo(e, CURLINFO_RESPONSE_CODE, (void **)&http_status);
-
-	int status = rgw_http_error_to_errno(http_status);
-	int result = msg->data.result;
-	finish_request(req_data, status);
+        CURL *e = msg->easy_handle;
+        rgw_http_req_data *req_data;
+        curl_easy_getinfo(e, CURLINFO_PRIVATE, (void **)&req_data);
+        
+        curl_multi_remove_handle((CURLM *)multi_handle, e);
```

Comment:
is this call needed for the LOW_SPEED options to work, or was there a separate motivation for it?

Reply:
This is not needed for the options. It's recommended if we use "curl_multi_perform". The handle should be removed from multi_handle before finished, though there's only one handle in multi_handle here...

Reply:
@cbodley @zhangsw my cloud sync branch gets rid of the whole synchronous api, so any change in `process_requests` is pretty moot

Reply:
@yehudasa OK, I've removed it.